src/csharp/Microsoft.Spark/TaskContext.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Spark
+{
+    /// <summary>
+    /// TaskContext stores information related to a task.
+    /// </summary>
+    internal class TaskContext
+    {
+        internal TaskContext()
+        {
+            TaskContextHolder.Set(this);
+        }
```

Comment:
@imback82 do you prefer this to be set here or at the site where the TaskContext object gets created.

Reply:
I prefer the latter to be explicit.

Reply:
Registered the `TaskContext` with `TaskContextHolder` in `PayloadProcessor`